datadisplay.cpp

```diff
@@ -58,11 +58,17 @@ void DataDisplay::displayData(const QByteArray &data)
         m_timestamp = QStringLiteral("[") + timestamp.toString(m_timestampFormat) + QStringLiteral("] ");
     }
 
+    // Store selection position before appending new data
+    QTextCursor cursor = textCursor();
```

Comment:
direct access to textCursor is no longer possible. Please change it to
`m_dataDisplay->textCursor()`
`setTextCursor()` is affected too.